pkg/agent/ocp.go

```diff
@@ -47,12 +63,66 @@ func NewClusterOpenShiftAPIClient(ctx context.Context, assetDir string) (*Cluste
 
 }
 
+// AreClusterOperatorsInitialized Waits for all Openshift cluster operators to initialize
+func (ocp *ClusterOpenShiftAPIClient) AreClusterOperatorsInitialized() (bool, error) {
+
+	failing := configv1.ClusterStatusConditionType("Failing")
+
+	operators, err := ocp.ConfigClient.ConfigV1().ClusterOperators().List(ocp.ctx, metav1.ListOptions{})
+	if err != nil {
+		return false, errors.Wrap(err, "Listing ClusterOperator objects")
+	}
+
+	operator := operators.Items[0]
+
+	if cov1helpers.IsStatusConditionTrue(operator.Status.Conditions, configv1.OperatorAvailable) &&
+		cov1helpers.IsStatusConditionFalse(operator.Status.Conditions, failing) &&
+		cov1helpers.IsStatusConditionFalse(operator.Status.Conditions, configv1.OperatorProgressing) {
+		return true, nil
+	}
```

Comment:
I'm concerned that we're copying-and-pasting a lot of logic here from the IPI wait-for commands that now has the opportunity to get out of sync. (Also, the approach there of setting a watch that gets events pushed from the k8s API seems superior to polling every 2(!) seconds as we're doing here.)
Why not just flip those functions to public and call them?

Reply:
For the last two I'll write up some stories about how to go about this. I think it is doable and probably time to create some sort of "common" package for both flows to use.

Reply:
Follow up story AGENT-313